src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -193,6 +194,19 @@ public DocumentDatabase(string name, RavenConfiguration configuration, ServerSto
             }
         }
 
+        public void SetIds(RawDatabaseRecord record)
+        {
+            DatabaseGroupId = record.Topology.DatabaseTopologyIdBase64;
+            ClusterTransactionId = record.Topology.ClusterTransactionIdBase64;
+            ShardedDatabaseId = record.ShardedDatabaseId;
+        }
+        public void SetIds(DatabaseRecord record)
+        {
+            DatabaseGroupId = record.Topology.DatabaseTopologyIdBase64;
+            ClusterTransactionId = record.Topology.ClusterTransactionIdBase64;
+            ShardedDatabaseId = record.ShardedDatabaseId;
```

Comment:
create private method that accepts topology and shardedDatabaseId, use it in both places or just remove those and use new method